### PR TITLE
ENG-17076: delete stale data when creating a new PBD, which should on…

### PR DESCRIPTION
**Pull request by rdykiel**
…ly occur on a CatalogUpdate

NOTE: this could be considered for V9.1

### CODE REVIEW

File: src/frontend/org/voltdb/export/ExportGeneration.java [hunk 1]
```diff
@@ -676,6 +676,8 @@ private void addDataSources(Table table, int hostId,
                     }
                     final String key = table.getTypeName();
                     if (!dataSourcesForPartition.containsKey(key)) {
+                        // If we're creating a NEW data source it MUST be via catalog update
```
Comment (manjujames) [inline src/frontend/org/voltdb/export/ExportGeneration.java]: Isn't this also called when the server starts up with an existing catalog on a recover?

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java [hunk 2]
```diff
@@ -1353,6 +1360,11 @@ private Builder(Builder<?> builder, M extraHeader, BinaryDequeSerializer<M> seri
             return this;
         }
 
+        public Builder<M> create(boolean create) {
```
Comment (n3world) [inline src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: I have a slight preference for this to be named deleteExisting() or something that indicates that this option destroys an existing PBD.

File: src/frontend/org/voltdb/export/ExportGeneration.java (hunk 1 above)
Comment (rdykiel) [reply src/frontend/org/voltdb/export/ExportGeneration.java]: Tested the case, and on the recover you are creating the EDS from the AD file. My conversation with Walter came to the conclusion that new EDS instances are only created on catalog update. We are running system tests on the branch to validate that assumption.

File: src/frontend/org/voltdb/utils/PersistentBinaryDeque.java (hunk 2 above)
Comment (rdykiel) [reply src/frontend/org/voltdb/utils/PersistentBinaryDeque.java]: I shall rename to the suggestion.

**Pull request merged**